app/Policies/PreferencePolicy.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Policies;
+
+use App\Models\Preference;
+use App\Models\User;
+
+class PreferencePolicy
```

Comment:
Is it registered in `AuthServiceProvider.php` ?
Also you created policy but does not us it in controller.